apiserver/pinger_test.go

```diff
@@ -149,18 +148,11 @@ func waitForClock(c *gc.C, clock *testing.Clock) {
 }
 
 func checkConnectionDies(c *gc.C, conn api.Connection) {
-	attempt := utils.AttemptStrategy{
-		Total: coretesting.LongWait,
-		Delay: coretesting.ShortWait,
-	}
-	for a := attempt.Start(); a.Next(); {
-		err := pingConn(conn)
-		if err != nil {
-			c.Assert(err, gc.ErrorMatches, "connection is shut down", gc.Commentf("details: %s", errors.Details(err)))
-			return
-		}
+	select {
+	case <-conn.Broken():
```

Comment:
given that this is in "pinger_test.go" are we sure we aren't missing the coverage around "ping notices that the connection is broken"?

Part of this is almost definitely "make sure the test suite cleans up" which I think your patch is fine for.
I'm just not sure if it is also intending to cover the "Ping() notices when the connection dies".

Thoughts?